packages/appium/lib/main.js

```diff
@@ -172,13 +169,34 @@ async function main (args = null) {
     // otherwise parse from CLI
     args = parser.parse_args();
   }
+
+  const configResult = await readConfigFile(args.config);
+  // TBD: maybe just warn?
+  if (!_.isEmpty(configResult.errors)) {
+    throw new Error(`Errors in config file ${configResult.filepath}:\n ${configResult.reason ?? configResult.errors}`);
+  }
+
+  // merge config and apply defaults.
+  // the order of precendece is:
+  // 1. command line args
+  // 2. config file
+  // 3. defaults from config file
+  args = _.defaults(args, configResult.config?.server, getDefaultsFromSchema({prop: 'server'}));
+
   await logsinkInit(args);
 
+  if (configResult.config) {
+    const configFilepath = path.relative(process.cwd(), configResult.filepath);
+    logger.info(`Loaded Appium config from ${configFilepath.yellow}`);
+  }
+
   // if the user has requested the 'driver' CLI, don't run the normal server,
   // but instead pass control to the driver CLI
   if (args.subcommand === DRIVER_TYPE || args.subcommand === PLUGIN_TYPE) {
+    // TODO: extract driver/plugin-specific config from config file here, and merge
+    // it with the args.
     await runExtensionCommand(args, args.subcommand);
-    process.exit();
+    return;
```

Comment:
why not to return?
does this change have anything to do with the original topic?

Reply:
it doesn't.  I can revert it for this PR, but you shouldn't `process.exit()` while relying so heavily on async code.